core/indigo-core/molecule/meta_commons.h

```diff
@@ -38,29 +39,170 @@ namespace indigo
+
+    enum class KETVersionIndex : int
+    {
+        EMajor,
+        EMinor,
+        EPatch
+    };
+
+    using KETFontVal = std::variant<std::monostate, std::string, uint32_t>;
+
+    struct KETFontStyleStatus
+    {
+        KETFontStyleStatus() : offset(0), size(0), val(std::monostate{})
+        {
+        }
+
+        KETFontStyleStatus(std::size_t off, std::size_t sz, KETFontVal v) : offset(off), size(sz), val(v)
+        {
+        }
+
+        std::size_t offset;
+        std::size_t size;
+        KETFontVal val;
+    };
+
+    struct KETFontStyle
+    {
+        enum class FontStyle : int
+        {
+            ENone,
+            EBold,
+            EItalic,
+            ESuperScript,
+            ESubScript,
+            EFamily,
+            ESize,
+            EColor
+        };
+
+        KETFontStyle(const FontStyle fs, KETFontVal val = std::monostate{}) : _font_style(fs), _val(val)
+        {
+        }
+
+        KETFontStyle() : _font_style(FontStyle::ENone)
+        {
+        }
+
+        KETFontStyle(const KETFontStyle& other) : _font_style(other._font_style), _val(other._val)
+        {
+        }
+
+        operator int() const
+        {
+            return static_cast<int>(_font_style);
+        }
+
+        operator FontStyle() const
+        {
+            return _font_style;
+        }
+
+        FontStyle getFontStyle() const
+        {
+            return _font_style;
+        }
+
+        const KETFontVal& getVal() const
+        {
+            return _val;
+        }
+
+        std::optional<std::string> getString() const
+        {
+            if (auto str = std::get_if<std::string>(&_val))
+                return *str;
+            return std::nullopt;
+        }
+
+        std::optional<std::uint32_t> getUInt() const
+        {
+            if (auto val = std::get_if<uint32_t>(&_val))
+                return *val;
+            return std::nullopt;
+        }
+
+        void setFontStyle(FontStyle fs)
+        {
+            _font_style = fs;
+        }
+
+        void copyValue(const KETFontStyle& other)
+        {
+            _val = other._val;
+        }
+
+        void setValue(uint32_t val)
+        {
+            _val = val;
+        }
+
+        void setValue(const std::variant<std::monostate, std::string, uint32_t>& val)
+        {
+            _val = val;
+        }
+
+        bool hasValue() const
+        {
+            return _val.index() != 0;
+        }
+
+    private:
+        FontStyle _font_style;
+        KETFontVal _val;
+    };
+
     struct compareFunction
     {
-        bool operator()(const std::pair<int, bool>& a, const std::pair<int, bool>& b) const
+        bool operator()(const std::pair<KETFontStyle, bool>& a, const std::pair<KETFontStyle, bool>& b) const
         {
-            return a.second == b.second ? a.first < b.first : a.second < b.second;
+            return a.second == b.second ? static_cast<int>(a.first) < static_cast<int>(b.first) : a.second < b.second;
```

Comment:
Looks strange.
Better use `static_cast<int>(a.first.FontStyle)` or ad explicit function to get int value of font style

Reply:
fixed